vowpalwabbit/OjaNewton.cc

```diff
@@ -578,7 +578,7 @@ base_learner* OjaNewton_setup(arguments& arg)
 
   arg.all->weights.stride_shift((uint32_t)ceil(log2(ON->m + 2)));
 
-  learner<OjaNewton>& l = init_learner(ON, learn, predict, arg.all->weights.stride());
+  learner<OjaNewton, example>& l = init_learner(ON, learn, predict, arg.all->weights.stride());
```

Comment:
Does carrying the type information have value after init_learner?  Basically, I'm wondering if type erasure should happen inside of init_learner().

Reply:
Fixed:  set_* functions use E from class template when example type is needed.